tests/test_git.py

```diff
@@ -214,9 +216,13 @@ def test_get_ref(tmp_dir: TmpDir, scm: Git, git: Git):
             ): "ref: refs/heads/master",
         }
     )
+    scm.tag(["-a", "annotated", "-m", "Annotated Tag"])
 
     assert init_rev == git.get_ref("refs/foo/bar")
     assert init_rev == git.get_ref("refs/foo/baz")
+    assert init_rev == git.get_ref("refs/tags/annotated")
+    # follow=False to replicate dvc.repo.experiments.show
+    assert scm.get_rev() == git.get_ref("refs/tags/annotated", follow=False)
```

Comment:
the comment isn't necessary, we can make it obvious from looking at the test that we are checking the annotated tag behavior for both `possible follow` values

(and we don't need the extra `get_rev()` call)

```suggestion
    assert init_rev == git.get_ref("refs/tags/annotated", follow=True)
    assert init_rev == git.get_ref("refs/tags/annotated", follow=False)
```